src/istgt_iscsi.c

```diff
@@ -7180,6 +7203,42 @@ istgt_iscsi_drop_all_conns(CONN_Ptr conn)
 	return (0);
 }
 
+// istgt_iscsi_is_existing_conn_alive returns 1 if
+// there are any existing connections that are alive
+// else return 0
+static int
+istgt_iscsi_is_existing_conn_alive(CONN_Ptr conn)
```

Comment:
any chance of race to reach here for two logins at same time?

Reply:
Yes, Since executing opcodes are done by [worker thread](https://github.com/openebs/istgt/blob/replication/src/istgt_iscsi.c#L6628-L6633)(which gets spun after a client connects to target) it is quite possible to have race b/w two different logins from different clients(/initiators). Since we are taking a [global lock-on on g_conns](https://github.com/mittachaitu/istgt/blob/handle_multiple_conns/src/istgt_iscsi.c#L7215) **I thought** it will solve race condition but the same can happen to another login request also if there is a delay in disconnection from other client(since we are returning non-retryable error)... and this behavior can repeat endlessly?

Reply:
can we introduce new state for this, to avoid race condition?

Reply:
Pushed this check into istgt_create_conn(there won't be any race)